### ISSUE TITLE
_ does not work in string interpolator

### ISSUE THREAD

**Issue by scf37** [lampepfl/dotty-feature-requests#318]

**Comment by dwijnand** [comment]
There's history here: https://github.com/scala/scala/pull/2793#issuecomment-22096120

**Comment by scf37** [comment]
Use case: Kotlin has very handy `it` keyword for functions with single anonymous parameter: `values.map { "\"$it\"" }`

It could be nice to have consistent `_` in Scala: `values.map(s"\"$_\"")`

**Comment by som-snytt** [comment]
I agree it would be nice, and I would have used the feature a few times over the years.

The specification is straightforward, as it is ordinary placeholder syntax.

Let me check:
```
def f[A](xs: Seq[A]) = xs.map("prefix" + _)
def f[A](xs: Seq[A]) = xs.map("prefix" + s"$_")
def f[A](xs: Seq[A]) = xs.map("prefix" + StringContext("", "").s(_))
```
which would normally be written
```
def f[A](xs: Seq[A]) = xs.map(s"prefix$_")
```
